LTP: fix test case sigrelse01 issue

shaikshavali1 opened the pull request:
This test case creates a child process using "FORK_OR_VFORK".
Fork behavior is not supported in sgx-lkl.
Hence, the below modifications are performed.
        1. Removed the code related to child process creation.
        2. Moved child and parent-related code/functionality
           to a single function "verify_sigrelse"
        3. Signal with ID 34 is skipped from testing.
           This signal ID is used for internal signals in
           the thread library of sgx-lkl-musl.
        4. Removed pipe handling code which is used for
           interprocess communication between parent and child.
        5. uClinux code is removed because exec is the problem.
        6. Issue (https://github.com/lsds/sgx-lkl/issues/598) is raised to support the loading of regular
           ELF binaries on a non-MMU architecture. This patch
           needs to be removed/reverted once Github issue https://github.com/lsds/sgx-lkl/issues/598
           is fixed.

shaikshavali1 committed:
LTP: fix testcase sigrelse01 issue

This test case creates a child process using "FORK_OR_VFORK".
Fork behavior is not supported in sgx-lkl.
Hence, the below modifications are performed.
        1. Removed the code related to child process creation.
        2. Moved child and parent-related code/functionality
           to a single function "verify_sigrelse"
        3. Signal with ID 34 is skipped from testing.
           This signal ID is used for internal signals in
           the thread library of sgx-lkl-musl.
        4. Removed pipe handling code which is used for
           interprocess communication between parent and child.
        5. uClinux code is removed because exec is the problem.
        6. Issue #598 is raised to support the loading of regular
           ELF binaries on a non-MMU architecture. This patch
           needs to be removed/reverted once Github issue #598
           is fixed.

hukoyu commented:
PR https://github.com/lsds/sgx-lkl/pull/676 is closed and resubmitted here. 
@davidchisnall can you review please? Thanks

davidchisnall commented on testcases/kernel/syscalls/sigrelse/sigrelse01.c:
```diff
@@ -733,6 +401,7 @@ int choose_sig(int sig)
 	case SIGALRM:
 	case SIGCANCEL:
 	case SIGTIMER:
+	case SIGSYNCCALL:
```
Signals 32, 33, and 34 are used by musl.  We should skip any signals in the `SIGRTMIN` to `SIGRTMAX` range.  These are reserved for userspace and may be used by libc.

shaikshavali1 replied:
Modified and submitted a new commit.

davidchisnall replied:
This code is still here.

davidchisnall commented:
@shaikshavali1  Please edit the commit message to refer to lsds/sgx-lkl#598, not it currently refers to an unrelated issue in the upstream repository.  You have now created a link between that unrelated issue and this PR, which is both impolite to the other project and unhelpful to us.

shaikshavali1 commented:
@davidchisnall, I removed the link.
As of now, this test skips 32, 33 &34. I will update the patch to skip all the signals between SIGRTMIN to SIGRTMAX range.

shaikshavali1 committed:
Address review comments

Signals 32, 33, and 34 are used by musl.
We should skip any signals in the SIGRTMIN to SIGRTMAX range.
These are reserved for userspace and may be used by libc.

davidchisnall commented on testcases/kernel/syscalls/sigrelse/sigrelse01.c:
```diff
@@ -116,23 +116,17 @@ extern int sighold(int __sig);
+// Needed for musl pthread implementation
 #define SIGCANCEL 32
 #define SIGTIMER 33
+#define SIGSYNCCALL 34
```
Should not be needed.

shaikshavali1 replied:
Removed the code

hukoyu commented:
@davidchisnall Shaik applied your feedback. Does that look good now? Is it ready to be approved?

davidchisnall reviewed (commented):
I apparently failed to hit the submit button last night. Please consider this approved when the last two co,meets are addressed.

shaikshavali1 committed:
Address the review comments

hukoyu commented:
@davidchisnall  Shaik addressed all your feedback. If it looks good, can you approve it? Thanks

hukoyu reviewed (approved):
LGTM
